File: Java/Shirt.java
/*
Saya Muhammad Rizki Revandi dengan NIM 2205027 mengerjakan Latihan Praktikum 2
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak
melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

// deklarasi kelas Shirt yang merupakan child class dari kelas Clothing
class Shirt extends Clothing
{
    // atribut - atribut privat untuk color dan sleeveType
    private String color;
    private String sleeveType;

    // konstruktor
    public Shirt()
    {
        this.color = "";
        this.sleeveType = "";
    }

    // konstruktor dengan parameter color dan sleeveType
    public Shirt(String color, String sleeveType)
    {
        this.color = color;
        this.sleeveType = sleeveType;
    }

    // get Color
    public String getColor()
    {
        return this.color;
    }

    // set Color
    public void setColor(String color)
    {
        this.color = color;
    }

    // get SleeveType
    public String getSleeveType()
    {
        return this.sleeveType;
    }

    // set SleeveType
    public void setSleeveType(String sleeveType)
    {
        this.sleeveType = sleeveType;
    }
}

File: CPP/Shirt.cpp
/*
Saya Muhammad Rizki Revandi dengan NIM 2205027 mengerjakan Latihan Praktikum 2
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak
melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

// library yang digunakan
#include <iostream>
#include <string>
#include "Clothing.cpp"

// import standard namespace
using namespace std;

// deklarasi kelas Clothing yang merupakan child class dari kelas Product
class Shirt : public Clothing
{
    private:
    // atribut - atribut privat untuk color dan sleeveType
    string color;
    string sleeveType;

    public:
    // constructor
        Shirt()
        {
            // set size, material, dan gender ke default
            this->color = "";
            this->sleeveType = "";
        }

        Shirt(string color, string sleeveType)
        {
            this->color = color;
            this->sleeveType = sleeveType;
        }

        /* getrer dan setrer setiap atribut */
        string getColor()
        {
            return this->color;
        }

        void setColor(string color)
        {
            this->color = color;
        }

        string getSleeveType()
        {
            return this->sleeveType;
        }

        void setSleeveType(string sleeveType)
        {
            this->sleeveType = sleeveType;
        }

        // destructor
        ~Shirt()
        {
            
        }
};